Makefile

```diff
@@ -1,15 +1,66 @@
-PROJECT = manta
-COMPILE_FIRST = manta_handler
-DEPS = cowlib dlhttpc jsx
-include erlang.mk
-
-otp_release = $(shell erl -noshell -eval 'io:fwrite("~s\n", [erlang:system_info(otp_release)]).' -s erlang halt)
-otp_ge_17 = $(shell echo $(otp_release) | grep -q -E "^[[:digit:]]+$$" && echo true)
-ifeq ($(otp_ge_17),true)
-	otp_ge_18 = $(shell [ $(otp_release) -ge "18" ] && echo true)
+.PHONY: deps compile rel
+
+REBAR3_URL=https://s3.amazonaws.com/rebar3/rebar3
+
+# Default to rebar on PATH
+REBAR3 ?= $(shell which rebar3)
+
+# If no rebar3 on PATH, then try the current directory
+ifeq ($(REBAR3),)
+ifeq ($(wildcard rebar3),rebar3)
+REBAR3 = $(CURDIR)/rebar3
 endif
+endif
+
+DIALYZER_APPS = kernel stdlib erts sasl eunit syntax_tools compiler crypto
+DEP_DIR = "_build/lib"
+SHORTSHA = `git rev-parse --short HEAD`
+PKGOS ?= $(shell uname -s)
 
-ifeq ($(otp_ge_18),true)
-	ERLC_OPTS += -Dtime_correction=1
-	TEST_ERLC_OPTS += -Dtime_correction=1
+ifeq ($(PKGOS),SunOS)
```

Comment:
If rebar3 is still not found, there isn't a great error message, it just fails like:

```
$ make
compile
make: compile: No such file or directory
make: *** [compile] Error 1
```